src/CollapsibleField.js

```diff
@@ -32,46 +34,87 @@ class CollapseMenuAction extends Component {
 
 function CollapseMenu(props) {
   let {
-    uiSchema: {
-      collapse: {
-        icon: {
-          enabled = "glyphicon glyphicon-chevron-down",
-          disabled = "glyphicon glyphicon-chevron-right",
-          add = "glyphicon glyphicon-plus-sign"
-        } = {},
-        separate = true,
-        addTo,
-        wrapClassName = "lead",
-        actions = [],
-        classNames = "collapsibleHeading",
-        collapseDivStyles: {
-          textColor = "white",
-          background = "linear-gradient(to right, #0472B6, white)",
-          collapseGlyphColor = "white",
-          addGlyphColor = "white",
-          padding = "14px",
-          margin = "",
-          marginLeft = "-5px",
-          marginBottom = "5px",
-          zIndex = -1,
-          divCursor = "pointer",
-          addCursor = "copy"
-        } = {}
-      }
-    },
+    headerElementsSchemas,
+    uiSchema,
     formContext = {},
+    formData = {},
     onChange,
     onAdd,
     title,
     name,
-    collapsed
+    collapsed,
+    fields,
+    propsOnChange
   } = props;
 
+  const {
+    collapse: {
+      collapsibleHeaderElements = {},
+      icon: {
+        enabled = "glyphicon glyphicon-chevron-down",
+        disabled = "glyphicon glyphicon-chevron-right",
+        add = "glyphicon glyphicon-plus-sign"
+      } = {},
+      separate = true,
+      addTo,
+      wrapClassName = "lead",
+      actions = [],
+      classNames = "collapsibleHeading",
+      collapseDivStyles: {
+        textColor = "white",
+        background = "linear-gradient(to right, #0472B6, white)",
+        collapseGlyphColor = "white",
+        addGlyphColor = "white",
+        padding = "14px",
+        margin = "",
+        marginLeft = "-5px",
+        marginBottom = "5px",
+        zIndex = -1,
+        divCursor = "pointer",
+        addCursor = "copy"
+      } = {}
+    }
+  } = uiSchema;
+
   const handleAdd = event => {
     event.stopPropagation();
     onAdd(event);
   };
 
+  let headerElements = [];
+  let {
+    className: headerElementsWrapperClass = "header-element-wrapper"
+  } = collapsibleHeaderElements;
+
+  Object.keys(headerElementsSchemas).map(key => {
+    const fieldSchema = headerElementsSchemas[key];
+    const fieldName = getFieldName(fieldSchema.type);
+
+    if (fieldName) {
+      let FieldElement = fields[fieldName];
+      const fieldId = `${key}`;
+      const fieldUiSchema = uiSchema[key];
+      const fieldFormData = formData[key];
+
+      const elementOnChange = (value, options) => {
+        formData[key] = value;
+        propsOnChange(formData, options);
+      };
+
+      headerElements.push(
+        <FieldElement
+          formContext={formContext}
+          formData={fieldFormData}
+          idSchema={{ $id: fieldId }}
+          key={key}
+          onChange={elementOnChange}
+          schema={fieldSchema}
+          uiSchema={fieldUiSchema}
```

Comment:
Does this contain the ui:order property?

Reply:
The header elements themselves are ordered by collapse.collapsibleHeaderElements.elements array.  The fieldUiSchema will have any ui:orders required for the element